### ISSUE TITLE
Suggestion: Add in information in the expedition tabs for overdrum requirements

### ISSUE THREAD

**Issue by Tsubakura** [KC3Kai/KC3Kai#1690]
Heya,

The exped tab is a very handy feature, which allows players to check if they meet the requirements for an expedition or not. However, it is currently missing information regarding the requirements for overdrumming.

Overdrumming is when you equip more drums then necessary for an expedition, which grants you a chance  for Great Success. Overdrum requirement is reached when you reach 25% more than the initial required drums.
The expeditions where overdrumming is possible, are 21,24,37,38 and 40.
The following amount of drums are required for each expedition:
- Expedition 21 - 4+ drums
- Expedition 24 - 4+ drums
- Expedition 37 - 5+ drums
- Expedition 38 - 10+ drums
- Expedition 40 - 4+ drums

What I want to suggest, is to add another statement under the expeditions where overdrumming is possible. The statement is red/yellow if the requirement aren't met, just like the others.  I am suggesting yellow as an alternative color, since I also want to make it obvious that overdrumming is not required for the expedition to succeed, its only for increasing the GS chance.
Example for exp 21:
If requirements are not met for success: Overdrum: 4+(RED)
If requirements are met for success, but not for overdrum: Overdrum: 4+(YELLOW)
If requirements for overdrumming are met: 4+(GREEN)

Example for exp 24:
As Exp 24 does not require drums to succeed, the red statement doesnt exist.
If requirements are met for success, but not for overdrum: Overdrum: 4+(YELLOW)
If requirements for overdrumming are met: 4+(GREEN)

This is a way to increase awareness to newfaces that overdrumming is a thing and that exp 24 and 40 can be overdrummed as well. I would love to see this implemented. :)

**Comment by Javran** [comment]
thanks for the suggestion. I'm fully aware of the fact of overdrumming, just that personally I don't like it and it might not worth the effort to implement - overdrumming can do just a tiny bit better than an all-sparkled fleet (theoretically it's just the difference between 99% and 100%), but the precondition is complicated (it's not just the number of drums, but also I believe you need some number of sparkled ships in a fleet as well) and at the time that I implemented exped tab the exact precondition was still under investigation, so I wasn't willing to put any unconfirmed info on the panel.

but anyway, I guess that's exactly what exped tab is for, maybe just a smaller indicator, or some tooltip when mouse moves over the "GS" toggle will do just fine.

**Comment by Tsubakura** [comment]
That would probably be a next step actually. There are 2 things we already know:

1. Every ship you sparkle gives you a ~19% chance of GS, ie. 5 sparkled ships gives you a ~95% chance of GS. You are still however, required to sparkle all the ships in the expedition fleet if the expedition does not support overdrumming.
2. Overdrumming in combination with 4 sparkled ships has been a guaranteed GS and I have yet to see one fail. 3 sparkled ships however, have seen to fail if in combination with overdrumming. There is enough data to conclude that the GS increase from overdrum is around ~24%, which is already a lot.
3. Expeditions where overdrumming is supported, does not require full sparkled fleet for GS to work at all. In fact, even sparkling 1 ship without overdrumming allows you a chance to GS, unlike the other expeditions.

Well, thats where we wikia and the #strategy boys come in if you want this to be analyzed further, but this is just to show that it's pretty well established as to how GS works with overdrumming.

**Comment by Javran** [comment]
@Tsubakura thanks again for the info. yep, I think a success rate on exped panel will be appropriate.

for me I always send all sparkled fleet to expeditions (except short ones like exped#2,3,6), and even then very occasionally I'll get just a normal success on some of them - turns out all of them are overdrumming expeditions (like the following) and I was too lazy to equip more drums:

![2016-12-25_142634_3840x1080_scrot](https://cloud.githubusercontent.com/assets/1096354/21472557/47511082-caae-11e6-95fb-1b80bdf117cd.jpg)


so somehow I believe the game forces you to do overdrumming on certain expeditions regardless of how sparkled your fleet is. if just having few more drums can rule out the last bit of uncertainty, it definitely worth a try.

**Comment by Tsubakura** [comment]
I'm not actually sure whether 6 sparkled ships being capped at 99% only applies to overdrum expeditions, since I have heard reports of it happening as well on other expeditions, so I'll have to dig around.

However, since overdrumming basically saves you 2 ships worth of sparkles, its a good thing to include regardless. 😄

**Comment by Javran** [comment]
@Tsubakura that's also what I've learned from wikiwiki, wikia and many other sources, however, I've previously examined my own long expedition history but couldn't find a single example of normal success on non overdrumming-expeditions.. either I'm lucky or wiki contributors want to be conservative on their statements.

**Comment by newttk** [comment]
Hello,
I'm new here but really like your project.

I never find the time the sparkle any ship, so I just use fully overdrummed fleets ( expect the small amount of daihatsus I have.) With this I've a roughly 50% GS rate. So this feature may be interesting for other busy/lazy TTKs. Normally I run exp 5,21 and 37/38 and either 21 and 3x have roughly 50% GS.

Also from my point of view, the drums seam to add (much) more GS-rate then noted above.

If i remember correctly, the GS-rate did not rise much after the first three/four excess drums. But as I have them, I just mount them (for maybe some extra percent of GS-rate), 

![image](https://cloud.githubusercontent.com/assets/24959150/21718943/1b1a6f4a-d45e-11e6-8208-6cb4b021f42d.png)

![image](https://cloud.githubusercontent.com/assets/24959150/21718965/498026c2-d45e-11e6-948f-8765a9e137ca.png)

Thanks

**Comment by Javran** [comment]
will be working on this. but now I want to abandon the idea of adding a GS rate on panel, unless we have clear answers to the following questions:

- how does the number of sparkled ships affect success rate? (ignoring overdrum expeditions)

    Many sources say each adds 19% GS rate and it's capped at a number very close to 100%. but [here (Chinese)](http://bbs.ngacn.cc/read.php?tid=8314468) I find some experiment result on ngacn that disproves this claim: the author ran exped 3 for 208 times with 3 sparkled ship, and 145 of them were GS. 145/208=69.7% is not even close to the estimated 19%x3=57%. but the claim seems to be accurate about expedition with 4-5 sparked ships (~76% for 4 sparkled-fleet, ~95% for 5 sparkled-fleet)

- same question as the first one, but for overdrum expeditions

    basically you can achieve ~50% even with no sparked ship. but how exactly does the number of drums and the number of sparkled ships affect GS rate?

- I found the following claim on [wikiwiki](http://wikiwiki.jp/kancolle/?%B1%F3%C0%AC):

> 耐久の損耗によって大成功になりにくくなる｡大成功を狙う場合は要回復｡

If I understand this correctly, damaged ships will lower GS rate. this makes the GS rate estimation even harder. (unless, say we can find a formula relating GS rate and damaged ships)

**Comment by Javran** [comment]
implemented. closing.

**Comment by nonsensed** [comment]
@Javran  If there is no reliable info on how much drums increase success rate, should the % display be disabled for expedition 21/37/38 (and maybe 24/40) to prevent this?
![Image](http://i.imgur.com/CsLF355.png)
 In the image it would be a 99-100% GS rate.

**Comment by Javran** [comment]
@nonsensed I did forget about this case. reopening this until I solved this (not right now though)

in addition, it doesn't make sense when the requirements are not met while you have a non-zero GS rate (which implies success rate in some sense) so that'll be the second change I'm planning to do.

**Comment by Tsubakura** [comment]
Yeah. The original idea is to include a message stating that one can increase the GS chances through overdrumming, the GS rate is secondary which still needs to be investigated.

For now, I think it should be fine to only include the overdrumming part, until we learn more about the exact GS rates.

**Comment by fourinone41** [comment]
Just saw this, so just wanted to add what I know about GS chance in Vita version. In Vita, GS chance = `21 + 15*#sparkled + drumBonus + flagshipLevelBonus`.
`drumBonus` is 20 if sufficient overdrum, or -15 if insufficient.
`flagshipLevelBonus` is a value applied to only some expeditions, with no real pattern, and equal to `floor(sqrt(LVL) + LVL/10) - 5`. Don't really know which expeditions in browser version this applies to, if any, since Vita's expeditions are all different.

This would be consistent with that ngacn test: 21 + 3*15 = 66.
It also explains <100% GS chance for overdrum expeditions: if all ships are sparkled but overdrum is not met, 21 + 6*15 - 15 = 96.

Whether you think any of this is good enough to use for browser version is up to you.

**Comment by Javran** [comment]
@fourinone41 the vita version formula results in 81% for 4 sparkled fleet and 96% for 5 sparkled fleet. I think the precision is okay but still a bit different than the conclusion from experiments (especially 79% for 4 sparkled ships). as we are just considering more than 3 sparkled ships I think current approach is good enough. but still thanks for the note and we might go back and rethink about this in future with more experiment and conclusions available.
